Tech-Journal-ConsoleApp/Tech-Journal-ConsoleApp/EmailEntry.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Net;
+using System.Net.Mail;
+using System.Text;
+
+namespace Tech_Journal_ConsoleApp
+{
+    class EmailEntry
+    {
+        public const string Path = "c:\\temp\\appsettings.txt";
+
+        private bool _validEmail;
+        private string _toThisEmailAddress;
+        private string _emailUsername;
+        private string _emailPassword;
+
+
+        public EmailEntry()
+        {
+            _validEmail = false;
+        }
+
+        public bool CheckforEmailSettings()
+        {
+            if (!File.Exists(Path))
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        public void GenerateSenderEmailSettings()
+        {
+            using (var sw = File.AppendText(Path))
+            {
+                Console.WriteLine("Email Settings were not found. Creating file to store email settings.");
+                while (!_validEmail)
```

Comment:
Can you try doing
`while (IsValidEmailAddress(_emailUsername))` instead of using the variable?